src/nav/ModalNavBar.js

```diff
@@ -1,33 +1,77 @@
 import React from 'react';
-import { View, Text } from 'react-native';
+import { View, Text, StyleSheet } from 'react-native';
 import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/Foundation';
 
 import boundActions from '../boundActions';
 import { CONTROL_SIZE } from '../common/platform';
-import { styles } from '../common';
 import { BRAND_COLOR } from '../common/styles';
+import { styles, Touchable, SearchInput } from '../common';
 import NavButton from './NavButton';
 
+const navBarStyles = StyleSheet.create({
+  touchable: {
+    paddingLeft: 10,
+    paddingRight: 10,
+  },
+});
+
+
 class ModalNavBar extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      isSearchEnabled: false
+
+  };
+  }
 
   props: {
     nav: any,
     title: string,
-    popRoute: () => void,
-  }
+    isSearchBarApplied: boolean,
+    searchBarOnChange: () => void
+  };
+
+  state: {
+    isSearchEnabled: boolean
+  };
 
   render() {
-    const { nav, title, popRoute } = this.props;
+    const { isSearchEnabled } = this.state;
+    const { nav, title, popRoute, searchBarOnChange, isSearchBarApplied } = this.props;
     const textStyle = [styles.navTitle, nav.index > 0 && { marginRight: CONTROL_SIZE }];
 
+    const changeSearchEnableState = () => {
```

Comment:
If you define a function inside the render() method, a new function is created every time.

If you declare it as a method the `this` will work if instead of
```
somefunc() {
}
```
if you make it
```
somefunc = () => {
}
```
but inside the class as a method, then the `this` will be bound correctly.